core/compiler/src/test/scala/com/fortysevendeg/exercises/compiler/CodeSnippetSpec.scala

```diff
@@ -0,0 +1,149 @@
+package com.fortysevendeg.exercises
+package compiler
+
+import org.scalatest._
+
+class CodeSnippetSpec extends FunSpec with Matchers with CodeSnippetUtilities {
+
+  describe("code snippet extraction") {
+    it("should extract a one line snippet and trim all whitespace") {
+
+      val code = """
+       |/** This is an example exercise.
+       |  * What value returns two?
+       |  */
+       |def addOne(value: Int) = {
+       |  value + 1
+       |}
+       """.stripMargin
+
+      extractSnippet(code) should be("value + 1")
+
+    }
+
+    it("should extract a multiline snippet and trim consistent prefix whitespace") {
+
+      def check(code: String) = new {
+        def against(result: String) = extractSnippet(code.stripMargin) should be(result.stripMargin.trim)
+      }
+      info("all uniform")
+
+      check("""
+       |/** Example.
+       |  */
+       |def exampleExercise(value: Int) = {
+       |  println("avoid")
+       |  printlt("certain")
+       |  println("death")
+       |}""")
+        .against("""
+          |println("avoid")
+          |printlt("certain")
+          |println("death")
+          """)
+
+      info("extra space line 1")
+
+      check("""
+       |/** Example.
+       |  */
+       |def exampleExercise(value: Int) = {
+       |   println("avoid")
+       |  printlt("certain")
+       |  println("death")
+       |}""")
+        .against("""
+          | println("avoid")
+          |printlt("certain")
+          |println("death")
+          """)
+
+      info("extra space line 2")
+
+      check("""
+       |/** Example.
+       |  */
+       |def exampleExercise(value: Int) = {
+       |  println("avoid")
+       |   printlt("certain")
+       |  println("death")
+       |}""")
+        .against("""
+          |println("avoid")
+          | printlt("certain")
+          |println("death")
+          """)
+
+      info("extra space line 3")
+
+      check("""
+       |/** Example.
+       |  */
+       |def exampleExercise(value: Int) = {
+       |  println("avoid")
+       |  printlt("certain")
+       |   println("death")
+       |}""")
+        .against("""
+          |println("avoid")
+          |printlt("certain")
+          | println("death")
+          """)
+
+      info("various extra space")
+
+      check("""
+       |/** Example.
+       |  */
+       |def exampleExercise(value: Int) = {
+       |      println("avoid")
+       |    printlt("certain")
+       |     println("death")
+       |}""")
+        .against("""
+          |  println("avoid")
+          |printlt("certain")
+          | println("death")
+          """)
+
+    }
+
+  }
+
+}
+
+trait CodeSnippetUtilities {
+
+  val parser = ScalaExerciseParser()
```

Comment:
Not for this PR but we should inject dependencies via companion objects implicits so they can be overriden down the road if needed.